src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -260,25 +260,13 @@ public async Task<ListResponseModel<BulkDeleteResponseModel>> BulkDeleteAsync([F
             throw new NotFoundException();
         }
 
-        // Ensure all secrets belong to the same organization.
-        var organizationId = secrets.First().OrganizationId;
-        if (secrets.Any(secret => secret.OrganizationId != organizationId) ||
-            !_currentContext.AccessSecretsManager(organizationId))
+        var authorizationResult = await _authorizationService.AuthorizeAsync(User, secrets, BulkSecretOperations.ReadAll);
```

Comment:
could you add integration test coverage for this ?
we already have two, but i think we should also have the edge cases, specifically:

1. First condition (line 258):
    1. no secret id provided (empty list), covers `secrets.Any()`
    2. secret id's provided but not all found, covers `secrets.Count != request.Ids.Count()`
2. Second condition (line 264) - Authorization did not succeed due to:
    1. Secrets not in the same organization
    2. Secret to which user does not have Read access

Also the success test only checks for the count, but not for content in the response, which is not good.

Reply:
Added in https://github.com/bitwarden/server/pull/4099/commits/76f3e9ad4e950908bfadce0e28d20f5cc2de0efc